### DISCUSSION TITLE
Installing both atom and pulsar plug-ins failed

### DISCUSSION THREAD

**Discussion by hanbing5210** [pulsar-edit/.github#237] [Community Package Help]
Does apm require special configuration to install plug-ins for pulsar after atom and pulsar are installed at the same time？

**Comment by savetheclocktower** [comment]
No. Can you tell us exactly what failed? What errors are you getting? Also, are you using `ppm` (or `pulsar -p`) on the command line instead of `apm`?

**Reply by hanbing5210** [reply]
An error occurred when ppm search was executed：  Client network socket disconnected before secure TLS connection was established

**Reply by savetheclocktower** [reply]
Are you able to install packages from inside Pulsar's settings? Or by clicking the “Install” button on package listings on https://web.pulsar-edit.dev/ ?

**Reply by hanbing5210** [reply]
The package on page （https://web.pulsar-edit.dev/） can be found but clicking the install button does not respond

**Reply by savetheclocktower** [reply]
And if you go to the “Install” pane in Pulsar's settings and search for your package?

**Reply by hanbing5210** [reply]
![image](https://github.com/pulsar-edit/.github/assets/32626294/0ba6ed77-d06e-4010-a959-6d893f3b3556)

**Reply by savetheclocktower** [reply]
I'm out of ideas. Someone else might be able to help when they wake up tomorrow morning.

**Reply by confused-Techie** [reply]
Sorry for my lag.
But if you don't mind me asking, what country are you connecting from? I've seen this issue pop up from multiple countries that are affected by the Great Firewall, it seems for whatever reason, the Pulsar Package Registry is affected greatly by this filtering. For users that are affected, it seems the best solution has been to utilize some proxy to bypass the filtering of the Great Firewall. If you have access to such a proxy, you can configure it within PPM with the following [instructions](https://github.com/pulsar-edit/ppm#behind-a-firewall). 

Otherwise, if you are not behind the Great Firewall or some other filter (eg such as a work/school device), let me know since then this issue may be something else at play.

**Reply by hanbing5210** [reply]
china

**Comment by hanbing5210** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
